src/Server/GRPCServer.cpp

```diff
@@ -320,8 +321,20 @@ namespace
 
         Poco::Net::SocketAddress getClientAddress() const
         {
-            String peer = grpc_context.peer();
-            return Poco::Net::SocketAddress{peer.substr(peer.find(':') + 1)};
+            /// Returns a string like ipv4:127.0.0.1:55930 or ipv6:%5B::1%5D:55930
+            String uri_encoded_peer = grpc_context.peer();
+
+            if (uri_encoded_peer.starts_with("ipv4") || uri_encoded_peer.starts_with("ipv6"))
+            {
+                uri_encoded_peer = uri_encoded_peer.substr(uri_encoded_peer.find(':') + 1);
```

Comment:
This `find(':')` looks unnecessary - the only acceptably position of `:` is right after the prefix. So it seems better to add `:` to the prefix and just skip the whole prefix instead of looking for `:`.